solr/kubedb_client_builder.go

```diff
@@ -0,0 +1,139 @@
+package solr
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	gerr "github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"kubedb.dev/apimachinery/apis/kubedb"
+
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/go-resty/resty/v2"
+	api "kubedb.dev/apimachinery/apis/kubedb/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KubeDBClientBuilder struct {
+	kc      client.Client
+	db      *api.Solr
+	url     string
+	podName string
+	ctx     context.Context
+	log     logr.Logger
+}
+
+func NewKubeDBClientBuilder(kc client.Client, db *api.Solr) *KubeDBClientBuilder {
+	return &KubeDBClientBuilder{
+		kc: kc,
+		db: db,
+	}
+}
+
+func (o *KubeDBClientBuilder) WithPod(podName string) *KubeDBClientBuilder {
+	o.podName = podName
+	return o
+}
+
+func (o *KubeDBClientBuilder) WithURL(url string) *KubeDBClientBuilder {
+	o.url = url
+	return o
+}
+
+func (o *KubeDBClientBuilder) WithLog(log logr.Logger) *KubeDBClientBuilder {
+	o.log = log
+	return o
+}
+
+func (o *KubeDBClientBuilder) WithContext(ctx context.Context) *KubeDBClientBuilder {
+	o.ctx = ctx
+	return o
+}
+
+func (o *KubeDBClientBuilder) GetSolrClient() (*Client, error) {
+	if o.podName != "" {
+		o.url = o.GetHostPath(o.db)
+	}
+	if o.url == "" {
+		o.url = o.GetHostPath(o.db)
+	}
+	if o.db == nil {
+		return nil, errors.New("db is empty")
+	}
+	config := Config{
+		host: o.url,
+		transport: &http.Transport{
+			IdleConnTimeout: time.Minute * 6,
```

Comment:
Why do we have 6 minutes (such long periods) of timeouts everywhere? and can we use a variable for it?